superset-frontend/src/dashboard/components/SliceHeader/index.tsx

```diff
@@ -239,6 +242,9 @@ const SliceHeader: FC<SliceHeaderProps> = ({
       <div className="header-controls">
         {!editMode && (
           <>
+            {SliceHeaderExtension && (
+              <SliceHeaderExtension slice={slice} dashboardId={dashboardId} />
```

Comment:
do we need the entire slice or just the id?

Reply:
You're right, if an extension needs entire slice, it can be pulled from redux store using just the id. Updated